File: src/main/java/com/hospital/service/AuthService.java
package com.hospital.service;

import com.hospital.dto.request.LoginRequest;
import com.hospital.dto.request.RegisterRequest;
import com.hospital.dto.response.AuthResponse;

/**
 * Contract for authentication operations.
 */
public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    AuthResponse refreshToken(String refreshToken);
}

File: frontend/src/services/authService.js
import api from './api';

export const authService = {
  register: (data)         => api.post('/auth/register', data),
  login:    (data)         => api.post('/auth/login',    data),
  refresh:  (refreshToken) => api.post('/auth/refresh',  null, { params: { refreshToken } }),
};
